### ISSUE TITLE
Feature request: local hostnames configurable for the native loader

### ISSUE THREAD

**Issue by while1malloc0** [percy/percy-capybara#5]
I'm currently using powder to separate some locally running apps into their own domains. This seems to break Percy's ability to recognize that they're being run locally, and causes the CSS upload to err. Manually adding each powder domain to the LOCAL_HOSTNAMES list in `native_loader.rb` seems to resolve the issue, but having a more robust configuration option would be preferable. I'd be happy to do a PR for this, but I wanted to make sure that it was an architectural change that the maintainers would want.

**Comment by fotinakis** [comment]
Hey @while1malloc0, interesting bug. What are powder domains and what do they look like?

Alternatively, a maintainable way for you to fix this might be to subclass `Percy::Capybara::Loaders:: NativeLoader` and override `LOCAL_HOSTNAMES`—but then we have to provide a mechanism for you to pass in or set the `loader`, which doesn't currently exist.  This would likely need to happen in `Percy::Capybara::Client` (https://github.com/percy/percy-capybara/blob/92c8a400f3c2752a6c5957b5271274b31de20fb7/lib/percy/capybara/client.rb#L21) and probably with some exposed interface in the top-level Percy::Capybara interface as well.

**Comment by while1malloc0** [comment]
Powder is just a wrapper for Pow, which is just a rack server. The domains themselves look like ordinary urls, though it's something of a convention to prefix them with 'local' as in 'local.appname.com'. I'm not sure if this convention is enforced at any level, which would make an approach like changing LOCAL_HOSTNAMES to include some sort of regex moot. 

The subclassing approach sounds reasonable, and it looks relatively easy to just accept an instance of a loader as part of the `Client.initialize` options hash and make the appropriate conditional in `Client.initialize_loader`. 

Also, apropos of nothing, https://github.com/percy/percy-capybara/blob/92c8a400f3c2752a6c5957b5271274b31de20fb7/lib/percy/capybara/client.rb#L46 has a minor misspelling 😄

**Comment by fotinakis** [comment]
Fixed by #6

**Comment by guilherme** [comment]
We're facing the same problem. We have 3rd party urls for assets (almost all images of our app don't come from localhost). One issue that I see of subclassing the Native Loader, is that if it changes the subclass might break. It feels for me that this behaviour could be added to the NativeLoader. If it had a way to add localhost hostnames dynamically it would be awesome. What do you guys think?

**Comment by fotinakis** [comment]
@guilherme are you having that problem on the latest versions of percy-capybara? I'd think this would be fixed by https://github.com/percy/percy-capybara/pull/12 (in the native loader), but I'm not sure if we totally understand the problem.

**Comment by mfazekas** [comment]
#12 only grabs assets with native loader from the same host as the requested one. 
@guilherme 
> We have 3rd party urls for assets

Does it mean something like `s3` assets? If those are public then percy backend can grab them too.

Ideally we'd have some api, so one could customize native loader.

```ruby
Percy::Capybara.use_loader(NativeLoader.loader_with_options(asset_hostnames: ['','',])
```

**Comment by guilherme** [comment]
In my case the services are not exposed to the public network, so the urls map to private hosts and so percy can't reach them.

I like your idea @mzfazekas

**Comment by fotinakis** [comment]
In 2.4.1 we now pass through loader_options and the ability to do `use_loader(:native)` so we could extend the native loader to accept `use_loader(:native, asset_hostnames: ['foo.dev'])` and merge the hostnames over the defaults, or something.

**Comment by guilherme** [comment]
that would be great @fotinakis

**Comment by guilherme** [comment]
I've added a PR  #23 please let me know your thoughts about it

**Comment by timhaines** [comment]
Thanks for the contribution @guilherme!  Merged as https://github.com/percy/percy-capybara/pull/24  :)

**Comment by fotinakis** [comment]
Released in gem v2.4.2
